2019-10-28  Thant Zin Oo  <classmfk@lnxsrv07.seas.ucla.edu>

	* fixed-timespec.h (timespec_cmp):
	Normalized return values to be -1, 0, or 1, depending
	on direct comparison operators (<, >, ==). Fixes overflow
	problems with timestamps very far in the past.
	(timespec_cmp):

--- lib/timespec.h	2005-09-21 23:05:39.000000000 -0700
+++ fixed-timespec.h	2019-10-28 22:42:56.203542000 -0700
@@ -45,8 +45,11 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  if (a.tv_sec < b.tv_sec)
+    return -1;
+  else if (a.tv_sec > b.tv_sec)
+    return 1;
+  return 0;
 }
 
 # if ! HAVE_DECL_NANOSLEEP
